app/controllers/eligibilities_controller.rb

```diff
@@ -5,6 +5,15 @@ class EligibilitiesController < ApplicationController
   #
   # Return all eligibilities sorted by name in ascending order.
   def index
+    if params[:category_id]
+      eligibilities = Eligibility.where(
+        "id in (select distinct eligibility_id from eligibilities_services where service_id in " \
+        "(select service_id from categories_services where category_id=?))", params[:category_id]
```

Comment:
Is there a reason you did this as a bunch of subqueries instead of a flatter set of joins? I think in Rails, you could even do:

```ruby
Eligibility.joins({ services: :categories })
  .where({ services: { categories: { id: params[:category_id] } } })
```